### PR TITLE
fix(pipeline): skip union-level imagery and EUDR for multi-region runs (#860)

### LINKED ISSUE #860: fix(pipeline): skip union-level imagery for multi-region runs, add multi_region flag to manifest
## Problem

`run_enrichment` always computes a union-level frame plan and mosaic registration using the bounding box of all AOIs combined. For a supply-chain portfolio with parcels in Brazil, Indonesia, and Côte d'Ivoire, this union bbox spans two continents. The resulting NDVI summary, mosaic tiles, and change detection numbers are geographically meaningless.

These union-level stats appear in the manifest and get rendered in the dashboard alongside valid per-AOI data — an auditor reviewing the compliance PDF will immediately challenge them.

## Fix

Add a `multi_region` detection heuristic to `run_enrichment`:

```python
def _is_multi_region(per_aoi_coords: list[dict]) -> bool:
    """True when AOI centroids span > 500 km — union imagery is not meaningful."""
    ...
```

When `multi_region=True`:
- Skip the union-level `_run_mosaic_ndvi_phase` and `_run_change_detection_phase`
- Skip `_run_eudr_phase` at the union level (each AOI gets its own via `_enrich_single_aoi`)
- Still run union-level weather (as a fallback note, clearly labelled)
- Set `results["multi_region"] = True` in the manifest

The 500 km threshold is a starting point — validate against real supply-chain datasets. Add a constant `MULTI_REGION_THRESHOLD_KM` to `treesight/constants.py`.

## Acceptance criteria

- [ ] `multi_region: true` present in the manifest when AOI centroids span > `MULTI_REGION_THRESHOLD_KM`
- [ ] Union NDVI stats and mosaic search IDs are `null` / absent in the manifest for multi-region runs (not misleading zeros)
- [ ] `per_aoi_enrichment` entries are complete and correct regardless of `multi_region`
- [ ] Single-region runs (all parcels within one country) are unaffected
- [ ] `MULTI_REGION_THRESHOLD_KM` constant added to `treesight/constants.py`
- [ ] Tests cover: single-region (unchanged), multi-region (union skipped), edge case at threshold boundary
- [ ] `make test` passes

## Dependencies

None. Prerequisite for compliance table (#C) to display reliable aggregate stats.

## Frontend impact

The evidence display module should suppress the union-level NDVI chart and timelapse when `manifest.multi_region === true`. Each AOI's data comes from `per_aoi_enrichment[i]` instead. This frontend change can ship with the compliance table issue.

### PR DESCRIPTION
**Pull request by Hardcoreprawn**
### Summary

Implements multi-region detection and gates union-level pipeline phases behind it.

**Changes:**
- `treesight/constants.py` — adds `MULTI_REGION_THRESHOLD_KM = 500.0`
- `treesight/pipeline/enrichment/runner.py` — adds `_is_multi_region` helper (haversine centroid distance); `run_enrichment` skips union-level mosaic/NDVI, change detection, and EUDR phases when AOIs span >500 km; sets `multi_region` flag in manifest
- `tests/test_enrichment_runner.py` — new `TestMultiRegionDetection` and `TestMultiRegionOrchestrator` test classes covering single/multi-region, edge cases, union phase skipping, and per-AOI pass-through

**Why:** Union-level stats are meaningless/misleading when parcels span continents (e.g. Brazil + Indonesia). Skipping them reduces wasted compute and avoids producing incorrect EUDR evidence.

Closes #860.

---

- [x] All tests pass (`make test` — 1840 passed, 28 skipped)
- [x] Lint/format clean (`ruff check`, `ruff format`)
- [x] No new security surface

### CODE REVIEW

File: treesight/pipeline/enrichment/runner.py [hunk 1]
```diff
@@ -570,6 +571,49 @@ def _enrich_single_aoi(
...
+        lats = [c[1] for c in coords]
+        return sum(lons) / len(lons), sum(lats) / len(lats)
+
+    def _haversine_km(lon1: float, lat1: float, lon2: float, lat2: float) -> float:
+        r = 6_371.0  # Earth radius in km
+        phi1, phi2 = math.radians(lat1), math.radians(lat2)
+        dphi = math.radians(lat2 - lat1)
+        dlam = math.radians(lon2 - lon1)
+        a = math.sin(dphi / 2) ** 2 + math.cos(phi1) * math.cos(phi2) * math.sin(dlam / 2) ** 2
+        return 2 * r * math.asin(math.sqrt(a))
```
[the remark this answers is not in the document]
Comment (Hardcoreprawn) [reply treesight/pipeline/enrichment/runner.py]: Fixed in 991dca6. `EARTH_RADIUS_KM = EARTH_RADIUS_M / 1_000.0` is now defined in `treesight/constants.py` immediately after `EARTH_RADIUS_M`, and `_haversine_km()` imports and uses it instead of the inline `6_371.0`.

File: treesight/pipeline/enrichment/runner.py [hunk 2]
```diff
@@ -570,6 +571,49 @@ def _enrich_single_aoi(
...
+    still produced via the per-AOI fan-out.
+    """
+    import math
+
+    if len(per_aoi_coords) < 2:
+        return False
+
+    def _centroid(coords: list[list[float]]) -> tuple[float, float]:
+        lons = [c[0] for c in coords]
+        lats = [c[1] for c in coords]
```
[the remark this answers is not in the document]
Comment (Hardcoreprawn) [reply treesight/pipeline/enrichment/runner.py]: Fixed in 991dca6. `_centroid()` now strips the closing point before averaging: `pts = coords[:-1] if len(coords) > 1 and coords[0] == coords[-1] else coords`. This prevents the origin vertex from being double-weighted in centroid calculations for ring polygons.

File: treesight/pipeline/enrichment/runner.py [hunk 3]
```diff
@@ -654,34 +698,45 @@ def run_enrichment(
...
+        _run_change_detection_phase(
+            frame_plan,
+            ndvi_raster_paths,
+            output_container,
+            project_name,
+            timestamp,
+            storage,
+            results,
+            acc=acc,
+        )
```
[the remark this answers is not in the document]
Comment (Hardcoreprawn) [reply treesight/pipeline/enrichment/runner.py:742]: Fixed in 991dca6. The `determine_deforestation_free()` call is now gated on `if eudr_mode and not multi_region:`. For multi-region runs, union-level `change_detection` is absent, so a top-level determination would be unreliable. Per-AOI determinations in `per_aoi_enrichment` remain the authoritative signal.

File: tests/test_enrichment_runner.py [hunk 4]
```diff
@@ -638,3 +638,378 @@ def test_eudr_mode_runs_determination(self, mock_det):
...
+            project_name="test",
+            timestamp="20240101",
+            output_container="out",
+            storage=storage,
+            per_aoi_coords=per_aoi,
+        )
+
+        mock_mosaic.assert_not_called()
+        mock_change.assert_not_called()
+
```
[the remark this answers is not in the document]
Comment (Hardcoreprawn) [reply tests/test_enrichment_runner.py:1010]: Fixed in 991dca6. Added two new tests to `TestMultiRegionOrchestrator`:

- `test_multi_region_results_dict_lacks_union_outputs` — asserts that `search_ids`, `ndvi_search_ids`, `display_collections`, `ndvi_stats`, `ndvi_raster_paths`, and `change_detection` are all absent from the results dict for multi-region runs.
- `test_multi_region_eudr_skips_determination` — asserts that `determination` is absent from results when `eudr_mode=True` and AOIs are multi-region (since union `change_detection` is not available to base it on).
